elastic/apm-server
# beater: telemetry for tail-based sampling config

## Motivation/summary

Add some basic telemetry for tail-sampling config: if it's enabled, and the number of policies defined.

## Checklist


I have considered changes for:
~- [ ] documentation~
~- [ ] logging (add log lines, choose appropriate log selector, etc.)~
~- [ ] metrics and monitoring (create issue for Kibana team to add metrics to visualizations, e.g. [Kibana#44001](https://github.com/elastic/kibana/issues/44001))~

## How to test these changes

1. Run apm-server with monitoring enabled, and without tail-sampling enabled: you should see `apm-server.sampling.tail.enabled: false` in a `beats_state` monitoring doc.
2. Run apm-server with monitoring enabled, and tail-sampling enabled with two policies: you should see `apm-server.sampling.tail.enabled: true` and `apm-server.sampling.tail.policies: 2` in a `beats_state` monitoring doc.

## Related issues

https://github.com/elastic/apm-server/issues/4185
